src/oedisi/componentframework/system_configuration.py

```diff
@@ -152,17 +155,19 @@ def clean_model(self, target_directory="."):
             if proc.name() == "helics_broker":
                 proc.kill()
 
-    @validator("components")
+    @field_validator("components")
+    @classmethod
     def check_component_names(cls, components):
         "Check that the components all have unique names"
         names = set(map(lambda c: c.name, components))
         assert len(names) == len(components)
         return components
 
-    @validator("links")
-    def check_link_names(cls, links, values, **kwargs):
-        if "components" in values:
-            components = values["components"]
+    @field_validator("links")
```

Comment:
validator not returning anything

Reply:
my bad line was hidden

Reply:
Doesn't it return links on line 174 (previously 169)?